### DISCUSSION TITLE
How to hide cpu vm flags ?

### DISCUSSION THREAD

**Discussion by huangzhiran** [cloud-hypervisor/cloud-hypervisor#2876] [Q&A]
In clh, How to hide cpu vm flags ? Like QEMU `-cpu kvm=off` flags. Thanks !

**Comment by sboeuf** [comment]
Hi @huangzhiran 
We don't have the equivalent of `-cpu kvm=off` in Cloud Hypervisor. Do you have a requirement for this feature?

**Comment by huangzhiran** [comment]
Thanks for reply! I Passthrough the Nvidia device to clh VM

**Comment by huangzhiran** [comment]
But NVIDIA gpu driver cannot work when it detect running in a VM. The solution in QEMU is `-cpu kvm=off` to hide cpu vm flags. It. will be nice if Cloud Hypervisor support this feature.

**Reply by sboeuf** [reply]
Have you actually tested it? And did you observe an issue? I'm asking because we have our CI running VFIO with an NVIDIA Tesla card and it works fine. The NVIDIA driver is properly loaded and the card is usable.

**Reply by huangzhiran** [reply]
I tested with a GeForce RTX 2070 card. NVIDIA limit this kind of game card run for server purpose, and NVIDIA driver can not load when it detect running within vm with a game card.

**Reply by sboeuf** [reply]
Ok I see. That's a shame the NVIDIA driver behaves this way :-/

**Reply by sboeuf** [reply]
I've opened https://github.com/cloud-hypervisor/cloud-hypervisor/issues/2879 to summarize the issue and how we could solve it. Let's wait for @rbradford's input and if he's okay with this, we'll fix it quickly.

**Reply by rbradford** [reply]
> Ok I see. That's a shame the NVIDIA driver behaves this way :-/

This is a business decision on their part. They only supported device passthrough on their enterprise Quadro cards not their consumer cards. However they have changed this recently: https://www.techradar.com/uk/news/nvidia-finally-switches-on-geforce-gpu-passthrough to permit it on their consumer products.

**Reply by sboeuf** [reply]
That's good, but only about Windows, and only in beta so far. Do you think they're gonna apply this to Linux as well? And what does that mean for us in the meantime?

**Reply by rbradford** [reply]
> That's good, but only about Windows, and only in beta so far. Do you think they're gonna apply this to Linux as well? And what does that mean for us in the meantime?

First hit for "linux geforce vfio" on Google: https://www.reddit.com/r/linux_gaming/comments/mgh80t/nvidia_now_officially_supporting_gpu_passthrough/

**Reply by sboeuf** [reply]
Cool! Let's close this discussion/issue then :)
